specs/interop.md

```diff
@@ -0,0 +1,188 @@
+| Remote Chain  | A blockchain that includes a relaying message |
+| Initiating Message | A transaction submitted to a local chain that authorizes execution on a remote chain |
+| Relaying Message | A transaction submitted to a remote chain that corresponds to an initiating message |
+| Cross Chain Message | The cumulative execution and side effects of the initiating message and relaying message |
+| Interop Set | The set of chains that originate initiating transactions where the relaying transactions are valid |
+
+A total of two transactions are required to complete a cross chain message. The first transaction is submitted
+to the local chain which authorizes execution on a remote chain. The second transaction is submitted to a
+remote chain, where the block builder SHOULD only include it if they are certain that the first transaction was
+included in the local chain.
+
+The term "block builder" is used interchangeably with the term "sequencer" for the purposes of this document but they need not be the same
+entity in practice.
+
+## The Interop Set
+
+The interop set defines the set of chains that a remote chains allows as local chains. Another way of
+saying it is that the interop set defines the set of initiating messages that are valid for a relaying
+message to be included. A relaying message MUST have an initiating message that is included in a chain
+in the interop set.
+
+The interop set is defined by chain id. Since it is impossible to enforce uniqueness of chain ids,
+social consensus MUST be used to determine the chain that represents the canonical chain id. This
+particularly impacts the block builder as they SHOULD use the chain id to assist in validation
+of relaying messages. 
+
+### Chain ID
+
+The concept of a chain id was introduced in [EIP-155](https://eips.ethereum.org/EIPS/eip-155) to prevent
+replay attacks between chains. This EIP does not specify the max size of a chain id, although
+[EIP-2294](https://eips.ethereum.org/EIPS/eip-2294) attempts to add a maximum size. Since this EIP is
+stagnant, all representations of chain ids MUST be the `uint256` type.
+
+## Interop Network Upgrade
+
+The interop network upgrade timestamp defines the timestamp at which all functionality in this document is considered
+the consensus rules for an OP Stack based network. On the interop network upgrade block, a set of deposit transaction
+based upgrade transactions are deterministically generated by the derivation pipeline in the following order:
+
+- L1 Attributes Transaction calling `setL1BlockValuesEcotone`
+- User deposits from L1
+- Network Upgrade Transactions
+  - L1Block deployment
+  - Update L1Block Proxy ERC-1967 Implementation Slot
+
+The execution payload MUST set `noTxPool` to `false` for this block.
+
+The exact definitions for these upgrade transactions are still to be defined.
+
+## Derivation Pipeline
+
+The derivation pipeline enforces invariants on safe blocks that include relaying transactions.
```

Comment:
@protolambda @tynes 

General thought that popped up with safe head progression now become async for chain dependencies.

When message inclusion was a apart of the derivation pipeline, a great way to scale interop to tens of chains was through light clients. The verifier would only need to a depth of the interop peers relative to the max time an which a remote message would be included. This reduces resource constraints substantially

However now that message inclusion is up the user to send the relay message, I believe verifiers must now run archival nodes of all peers. Unless i'm missing something in my reasoning, are we okay with this tradeoff. If we want to scale interop to > 10 chains which was a goal previously, I think this is a problem

Reply:
> I believe verifiers must now run archival nodes of all peers.

Can you explain why this is the case? I do not believe it is the case. Verifiers need logs, not historical state

Reply:
a message can be replayed at any time. So even if we prune the storage data, we still need to permanently retain all emitted outbox logs from all remotes

Reply:
Logs are part of history, not state. You do not need an archive node to access them

Reply:
yea that's right we dont an archive node but we have growing storage requirements with this design that scales poorly. the state bloat is pushed into permanent receipts that need to be collected and stored